### DISCUSSION TITLE
communicate with OTBR by socket

### DISCUSSION THREAD

**Discussion by Leo2442926161** [openthread/openthread#8517] [Q&A] [answered]
Dear all,
Is there any documentation describes how C language communicate with OTBR?
I saw there's a picture, it's seems that they can communicate by socket? is there any lib that can be called by C language  API? Thank you.
![image](https://user-images.githubusercontent.com/30258768/206835171-fe09f9de-480a-4368-aa8d-70387a9f6fe3.png)

**Comment by jwhui** [comment, marked as answer]
I'm not sure what you mean by "communicate with OTBR". The OpenThread Border Router operate as an IPv6 router. In most cases, you do not communicate directly with the router, but devices that are reachable via the OpenThread Border Router.

Because Thread supports standard IPv6 communication, communicating via Thread can be done via standard sockets APIs. You should be able to find a number of guides on the web that provide examples of communicating via network sockets.
